internal/testutil/pkgdata.go

```diff
@@ -1,152 +1,327 @@
+			{name: "./home/"},
+			{name: "./lib/"},
+			{name: "./lib/systemd/"},
+			{name: "./lib/systemd/system/"},
+			{name: "./lib/systemd/system/motd-news.service"},
+			{name: "./lib/systemd/system/motd-news.timer"},
+			{name: "./proc/"},
+			{name: "./root/", mode: 00700},
+			{name: "./run/"},
+			{name: "./sbin/"},
+			{name: "./sys/"},
+			{name: "./tmp/", mode: 01777},
+			{name: "./usr/"},
+			{name: "./usr/bin/"},
+			{name: "./usr/bin/hello", mode: 00775},
+			{name: "./usr/games/"},
+			{name: "./usr/include/"},
+			{name: "./usr/lib/"},
+			{name: "./usr/lib/os-release"},
+			{name: "./usr/sbin/"},
+			{name: "./usr/share/"},
+			{name: "./usr/share/dict/"},
+			{name: "./usr/share/doc/"},
+			{name: "./usr/share/doc/base-files/"},
+			{name: "./usr/share/doc/base-files/copyright"},
+			{name: "./usr/share/info/"},
+			{name: "./usr/share/man/"},
+			{name: "./usr/share/misc/"},
+			{name: "./usr/src/"},
+			{name: "./var/"},
+			{name: "./var/backups/"},
+			{name: "./var/cache/"},
+			{name: "./var/lib/"},
+			{name: "./var/lib/dpkg/"},
+			{name: "./var/lib/misc/"},
+			{name: "./var/local/", mode: 02775},
+			{name: "./var/lock/", mode: 01777},
+			{name: "./var/log/"},
+			{name: "./var/run/"},
+			{name: "./var/spool/"},
+			{name: "./var/tmp/", mode: 01777},
+			{name: "./etc/os-release", linkname: "../usr/lib/os-release"},
+		},
+	},
+}
+
+//go:embed all:pkgdata
+var pkgdataFS embed.FS
+
+func (pkg *pkgDef) init(pkgName string) error {
+	if pkg.tarTemplate.Format == 0 {
+		pkg.tarTemplate = defaultTarTemplate
+	}
+	if pkg.sourceDir == "" {
+		pkg.sourceDir = "pkgdata/" + pkgName
+	}
+	if pkg.readSource == nil {
+		pkg.readSource = func(pkg *pkgDef, path string) ([]byte, error) {
+			localPath := filepath.Join(pkg.sourceDir, path)
+			return pkgdataFS.ReadFile(localPath)
+		}
+	}
+	switch pkg.compressExt {
+	case "gz", "xz", "zst":
+	case "":
+		pkg.compressExt = "zst"
+	default:
+		return fmt.Errorf("unknown compression: %s", pkg.compressExt)
+	}
+	return nil
+}
+
+func (pkg *pkgDef) writeTarEntry(tw *tar.Writer, entry *tarEntry) error {
+	var err error
+	header := entry.header
+
+	if header == nil {
+		newHeader := pkg.tarTemplate
+		header = &newHeader
+
+		header.Typeflag = tar.TypeReg
+		header.Name = entry.name
+		header.Linkname = entry.linkname
+		header.Mode = entry.mode & 01777
+
+		if entry.linkname != "" {
+			// use entry.header for hardlinks
+			header.Typeflag = tar.TypeSymlink
+		} else if entry.name[len(entry.name)-1] == '/' {
+			header.Typeflag = tar.TypeDir
+		}
+
+		// use entry.header or bits above 1777 for zero mode
+		if entry.mode == 0 {
+			switch header.Typeflag {
+			case tar.TypeLink:
+				header.Mode = 0777
+			case tar.TypeDir:
+				header.Mode = 0755
+			default:
+				header.Mode = 0644
+			}
+		}
+	}
+
+	var content []byte
+
+	if header.Typeflag == tar.TypeReg || header.Typeflag == tar.TypeRegA {
+		content = entry.content
+
+		if content == nil {
+			sourcePath := entry.source
```

Comment:
I'm now inlining the content in package definition. However, I still include base-files and read the content via a helper function (`readPkgdataFile`), as we have a lot of tests that depend on various files in base-files and their content (via hashes). Changing all the tests for that doesn't sound like minimum amount of work. :-) I propose that from now on, we add new package data as you say, by inlining small amount of data directly, but we keep base-files data. Is that acceptable?

Reply:
As suggested in a different comment, include the new data and new tests, keep the old data and tests as-is, port the old tests to the new inlined data in a separate PR. Let's please not add unwanted logic to avoid unwanted logic, as instead of moving forward that just adds further technical debt to be fixed.